### PR TITLE
Create safe-ethers-lib and safe-web3-lib packages

### LINKED ISSUE #152: Installing from npm does not install all needed dependencies
## Description

Installing with `npm install @gnosis.pm/safe-core-sdk` does not install needed dependencies like `@ethersproject/solidity` and `ethers`

## Environment

- Safe Core SDK version: 1.3.0
- Safe contract version: n/a
- Environment: browser

## Steps to reproduce

1. install with
2. try to use `Safe` and `Web3Adapter`: `import Safe, { Web3Adapter } from '@gnosis.pm/safe-core-sdk';`
3. Get error: `Module not found: Can't resolve 'ethers'`

## Expected result

I expect there should be no missing dependencies.

### PR DESCRIPTION
**Pull request by germartinez**
## What it solves
- Create new package: `safe-ethers-lib` (contains `EthersAdapter` and ethers-v5 typechain files)
- Create new package: `safe-web3-lib packages` (contains `Web3Adapter` and web3-v1 typechain files)
- Fixes issues with `web3` and `ethers` dependencies

Resolves #152 and #33

### CODE REVIEW

File: packages/safe-core-sdk/src/managers/contractManager.ts [hunk 1]
```diff
@@ -30,41 +54,31 @@ class ContractManager {
...
+      safeAddress,
       isL1SafeMasterCopy,
-      customContractAddress: safeAddress,
-      customContractAbi: customContracts?.safeMasterCopyAbi
+      customContracts
     })
-    if ((await ethAdapter.getContractCode(temporarySafeContract.getAddress())) === '0x') {
-      throw new Error('Safe Proxy contract is not deployed in the current network')
-    }
     const safeVersion = (await temporarySafeContract.getVersion()) as SafeVersion
```
Comment (iamacook) [inline packages/safe-core-sdk/src/managers/contractManager.ts:67]: Why does this need to be casted? Does `getVersion()` not have a return type?

File: packages/safe-core-sdk/src/managers/contractManager.ts [hunk 2]
```diff
@@ -82,6 +96,52 @@ class ContractManager {
...
+  }: GetMultiSendContractInstanceProps): Promise<MultiSendContract> {
+    const multiSendDeployment = getMultiSendContractDeployment(safeVersion, chainId)
+    const multiSendContract = await ethAdapter.getMultiSendContract({
+      safeVersion,
+      chainId,
+      singletonDeployment: multiSendDeployment,
+      customContractAddress: customContracts?.multiSendAddress,
+      customContractAbi: customContracts?.multiSendAbi
+    })
+    if ((await ethAdapter.getContractCode(multiSendContract.getAddress())) === '0x') {
```
Comment (iamacook) [inline packages/safe-core-sdk/src/managers/contractManager.ts:140]: You're checking the address against `0x` twice here. Is it done elsewhere as well? Might be worth creating a helper function?

Comment (iamacook) [reply packages/safe-core-sdk/src/managers/contractManager.ts:140]: It's also present in `packages/safe-core-sdk/src/managers/ownerManager.ts`

File: packages/safe-core-sdk/tests/utils/setupEthAdapter.ts [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
 import { ethers, web3 } from 'hardhat'
-import { EthAdapter, EthersAdapter, Web3Adapter } from '../../src'
 
 export async function getEthAdapter(signer: Signer): Promise<EthAdapter> {
   let ethAdapter: EthAdapter
   switch (process.env.ETH_LIB) {
     case 'web3':
       const signerAddress = await signer.getAddress()
-      ethAdapter = new Web3Adapter({ web3, signerAddress })
+      const web3AdapterConfig: Web3AdapterConfig = { web3, signerAddress }
```
Comment (iamacook) [inline packages/safe-core-sdk/tests/utils/setupEthAdapter.ts:12]: Not an issue, just out of curiosity: does this throw if the object is passed directly as a parameter?

File: packages/safe-ethers-lib/src/EthersAdapter.ts [hunk 4]
```diff
@@ -71,19 +65,14 @@ class EthersAdapter implements EthAdapter {
   getSafeContract({
     safeVersion,
     chainId,
-    isL1SafeMasterCopy,
+    singletonDeployment,
     customContractAddress
-  }: GetSafeContractProps): GnosisSafeContractEthers {
+  }: GetContractProps): GnosisSafeContractEthers {
     let contractAddress: string | undefined
```
Comment (iamacook) [inline packages/safe-ethers-lib/src/EthersAdapter.ts:71]: You could just assign this immediately via a ternary and not need the `if ... else`.

Comment (iamacook) [reply packages/safe-ethers-lib/src/EthersAdapter.ts:71]: There's actually a couple of other occurances in this file where you could do it too.

File: packages/safe-web3-lib/src/Web3Adapter.ts [hunk 5]
```diff
@@ -53,60 +48,53 @@ class Web3Adapter implements EthAdapter {
...
-      customContractAddress ?? safeSingletonDeployment?.networkAddresses[chainId]
+  }: GetContractProps): GnosisSafeContractWeb3 {
+    const contractAddress = customContractAddress ?? singletonDeployment?.networkAddresses[chainId]
     if (!contractAddress) {
       throw new Error('Invalid Safe Proxy contract address')
     }
     const safeContract = this.getContract(
       contractAddress,
-      customContractAbi ?? (safeSingletonDeployment?.abi as AbiItem[])
+      customContractAbi ?? (singletonDeployment?.abi as AbiItem[])
```
Comment (iamacook) [inline packages/safe-web3-lib/src/Web3Adapter.ts:61]: Does this still need to be casted after the refactor? Or should it wait until [this](https://github.com/gnosis/safe-core-sdk/issues/138).

Comment (iamacook) [reply packages/safe-web3-lib/src/Web3Adapter.ts:61]: The same goes for the other `AbiItem[]` cast below.

**Review by iamacook — COMMENTED** [review]
Muy bien! Just found a few things

File: packages/safe-core-sdk/tests/utils/setupEthAdapter.ts (hunk 3 above)
Comment (germartinez) [reply packages/safe-core-sdk/tests/utils/setupEthAdapter.ts:12]: It does not throw. Is just that sometimes I like to have the type handy

Comment (iamacook) [reply packages/safe-core-sdk/tests/utils/setupEthAdapter.ts:12]: Strictly typing = 💪

File: packages/safe-web3-lib/src/Web3Adapter.ts (hunk 5 above)
Comment (germartinez) [reply packages/safe-web3-lib/src/Web3Adapter.ts:61]: `SingletonDeployment` type from `safe-deployments` library defines the `abi` as a `any[]`

**Comment by germartinez** [comment]
I'm adding the changes in this PR, https://github.com/gnosis/safe-core-sdk/pull/158

**Review by iamacook — APPROVED** [review]
🚀

**Pull request merged**